pkg/daemon/controller/controller.go

```diff
@@ -253,13 +258,72 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	if err = sc.reconcileServices(cmData.Services); err != nil {
 		return ctrl.Result{}, err
 	}
+	if err = sc.reconcileEnvironmentVariables(cmData.EnvironmentVars, node); err != nil {
+		return ctrl.Result{}, err
+	}
 	// Version annotation is the indicator that the node was fully configured by this version of the services ConfigMap
 	if err = metadata.ApplyVersionAnnotation(sc.ctx, sc.client, node, desiredVersion); err != nil {
 		return ctrl.Result{}, fmt.Errorf("error updating version annotation on node %s: %w", sc.nodeName, err)
 	}
 	return ctrl.Result{}, nil
 }
 
+// systemEnvVarRegistryPath is where system level environment variables are stored in the Windows OS
+const systemEnvVarRegistryPath = `SYSTEM\CurrentControlSet\Control\Session Manager\Environment`
+
+// reconcileEnvironmentVariables ensures that all given environment variables are set as expected on the instance
+// There are 3 levels to Windows ENV variables: `Machine`, `User`, and `Process`. `Process` inherits from
+// `User`, which inherits from the top level `Machine`, and sub-processes inherit from their parent process.
+// In order to route all external node traffic through the proxy, we must set the proxy vars at the `Machine` level.
+func (sc *ServiceController) reconcileEnvironmentVariables(envVars map[string]string, node core.Node) error {
+	restartRequired := false
+	for name, expectedVal := range envVars {
+		// We cannot use the os.Setenv since it calls syscall.Setenv which, calls syscall.SetEnvironmentVariable
+		// As per Microsoft docs: "Calling SetEnvironmentVariable has no effect on the system environment variables"
+		registryKey, err := registry.OpenKey(registry.LOCAL_MACHINE, systemEnvVarRegistryPath, registry.ALL_ACCESS)
+		if err != nil {
+			return fmt.Errorf("unable to open Windows system registry key %s: %w", systemEnvVarRegistryPath, err)
+		}
+
+		actualVal, _, err := registryKey.GetStringValue(name)
+		if err != nil && err != registry.ErrNotExist {
+			return fmt.Errorf("unable to read environment variable %s: %w", name, err)
+		}
+		if actualVal != expectedVal {
+			err := registryKey.SetStringValue(name, expectedVal)
+			if err != nil {
+				// Do not log value as proxy information is sensitive
+				return fmt.Errorf("unable to set environment variable %s: %w", name, err)
+			}
+			restartRequired = true
+		}
+		if err = registryKey.Close(); err != nil {
+			return fmt.Errorf("unable to close Windows system registry: %w", err)
+		}
+	}
+	if restartRequired {
+		// Cordon and drain the Node before we restart the instance
+		drainHelper := nodeconfig.NewDrainHelper(sc.k8sclientset, klog.NewKlogr())
```

Comment:
where does `sc.k8sclientset` get initialized? we likely have to edit the `Options` struct and `setDefaults()` func